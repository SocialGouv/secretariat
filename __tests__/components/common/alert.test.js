import { render } from "@testing-library/react"

import Alert from "../../../src/components/common/alert"

it("renders warning alert", () => {
  const { container } = render(
    <Alert message="This is a warning alert." type="warning" />
  )
  expect(container).toMatchSnapshot()
})

it("renders info alert", () => {
  const { container } = render(
    <Alert message="This is a info alert." type="info" />
  )
  expect(container).toMatchSnapshot()
})

it("renders success alert", () => {
  const { container } = render(
    <Alert message="This is a success alert." type="success" />
  )
  expect(container).toMatchSnapshot()
})

it("renders error alert", () => {
  const { container } = render(
    <Alert message="This is a error alert." type="error" />
  )
  expect(container).toMatchSnapshot()
})
