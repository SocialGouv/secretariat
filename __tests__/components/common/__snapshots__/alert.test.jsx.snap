// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders error alert 1`] = `
<div>
  <div
    class="alert error"
  >
    <div
      class="icon"
    >
      <i
        class="ri-close-circle-fill"
      />
    </div>
    <div
      class="body"
    >
      <div
        class="message"
      />
    </div>
  </div>
</div>
`;

exports[`renders info alert 1`] = `
<div>
  <div
    class="alert info"
  >
    <div
      class="icon"
    >
      <i
        class="ri-information-fill"
      />
    </div>
    <div
      class="body"
    >
      <div
        class="message"
      />
    </div>
  </div>
</div>
`;

exports[`renders success alert 1`] = `
<div>
  <div
    class="alert success"
  >
    <div
      class="icon"
    >
      <i
        class="ri-checkbox-circle-fill"
      />
    </div>
    <div
      class="body"
    >
      <div
        class="message"
      />
    </div>
  </div>
</div>
`;

exports[`renders warning alert 1`] = `
<div>
  <div
    class="alert warning"
  >
    <div
      class="icon"
    >
      <i
        class="ri-alert-fill"
      />
    </div>
    <div
      class="body"
    >
      <div
        class="message"
      />
    </div>
  </div>
</div>
`;
